src/main/java/seedu/duke/modules/ModuleList.java

```diff
@@ -75,12 +80,17 @@ public void changeModuleGrade(String moduleCode, String grade) {
         try{
             Module toChange = getModule(moduleCode);
             toChange.setModuleGrade(grade);
+            System.out.println("Grade for " + moduleCode + " updated to " + grade);
+            assert toChange.getModuleGrade().equals(grade) : "Grade is not updated successfully";
+
         } catch (ModuleNotFoundException e){
-            System.out.println("Module not found in either list");
+            System.out.println("Module not found in list");
+        } catch (ModuleException e){
+            System.out.println(e.getMessage());
         }
     }
 
-    public double tallyGPA() {
+    public double tallyGPA() throws GpaException {
```

Comment:
Looking at how the GpaException is being throw, maybe you will want to rename it to a better name like NoModulesForGpaException GpaCalculationNullException or something like that

Reply:
I have renamed the exception to GpaNullException.